CMakeModules/EosioTester.cmake.in

```diff
@@ -10,125 +9,46 @@ set(EOSIO_VERSION "@VERSION_FULL@")
-   link_directories(${LLVM_LIBRARY_DIR})
-endif()
-
 set( CMAKE_CXX_STANDARD 17 )
 set( CMAKE_CXX_EXTENSIONS ON )
 set( CXX_STANDARD_REQUIRED ON )
 
 #adds -pthread. Ubuntu eosio.contracts build breaks without this flag specified
 set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
 set(THREADS_PREFER_PTHREAD_FLAG TRUE)
-find_package(Threads)
+
+include(CMakeFindDependencyMacro)
+
+set(LLVM_DIR @LLVM_CMAKE_DIR@)
+if (LLVM_DIR)
+   find_dependency(LLVM)
+endif()
+
+get_filename_component(BOOST_ROOT @Boost_INCLUDE_DIR@ DIRECTORY)
+find_dependency(Boost COMPONENTS date_time filesystem chrono iostreams)
+
+get_filename_component(OPENSSL_ROOT_DIR @OPENSSL_INCLUDE_DIR@ DIRECTORY)
+find_dependency(OpenSSL)
+
+find_dependency(Threads)
+find_dependency(ZLIB)
+
+include(${CMAKE_CURRENT_LIST_DIR}/tester-targets.cmake)
 
 if ( APPLE )
    set( CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_CXX_FLAGS} -Wall -Wno-deprecated-declarations" )
 else ( APPLE )
    set( CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_CXX_FLAGS} -Wall")
 endif ( APPLE )
 
-### Remove after Boost 1.70 CMake fixes are in place
-set( Boost_NO_BOOST_CMAKE ON CACHE STRING "ON or OFF" )
-set( Boost_USE_STATIC_LIBS ON CACHE STRING "ON or OFF" )
-find_package(Boost @Boost_MAJOR_VERSION@.@Boost_MINOR_VERSION@ EXACT REQUIRED COMPONENTS
-   date_time
-   filesystem
-   system
-   chrono
-   iostreams
-   unit_test_framework)
-
-find_library(libtester eosio_testing @CMAKE_INSTALL_FULL_LIBDIR@ NO_DEFAULT_PATH)
-find_library(libchain eosio_chain @CMAKE_INSTALL_FULL_LIBDIR@ NO_DEFAULT_PATH)
-find_library(libfc fc @CMAKE_INSTALL_FULL_LIBDIR@ NO_DEFAULT_PATH)
-find_library(libsecp256k1 secp256k1 @CMAKE_INSTALL_FULL_LIBDIR@ NO_DEFAULT_PATH)
-
-find_library(libff ff @CMAKE_INSTALL_FULL_LIBDIR@ NO_DEFAULT_PATH)
-find_library(libwasm WASM @CMAKE_INSTALL_FULL_LIBDIR@ NO_DEFAULT_PATH)
-find_library(libwast WAST @CMAKE_INSTALL_FULL_LIBDIR@ NO_DEFAULT_PATH)
-find_library(libir IR     @CMAKE_INSTALL_FULL_LIBDIR@ NO_DEFAULT_PATH)
-find_library(liblogging Logging @CMAKE_INSTALL_FULL_LIBDIR@ NO_DEFAULT_PATH)
-find_library(libruntime Runtime @CMAKE_INSTALL_FULL_LIBDIR@ NO_DEFAULT_PATH)
-find_library(libsoftfloat softfloat @CMAKE_INSTALL_FULL_LIBDIR@ NO_DEFAULT_PATH)
-get_filename_component(cryptodir @OPENSSL_CRYPTO_LIBRARY@ DIRECTORY)
-find_library(liboscrypto crypto "${cryptodir}" NO_DEFAULT_PATH)
-get_filename_component(ssldir @OPENSSL_SSL_LIBRARY@ DIRECTORY)
-find_library(libosssl ssl "${ssldir}" NO_DEFAULT_PATH)
-find_library(libchainbase chainbase @CMAKE_INSTALL_FULL_LIBDIR@ NO_DEFAULT_PATH)
-find_library(libbuiltins builtins @CMAKE_INSTALL_FULL_LIBDIR@ NO_DEFAULT_PATH)
-
-#Ubuntu build requires rt library to be specified explicitly
-if(UNIX AND NOT APPLE)
-  find_library(LIBRT rt)
-endif()
-
 set(EOSIO_WASM_RUNTIMES @EOSIO_WASM_RUNTIMES@)
 if("eos-vm-oc" IN_LIST EOSIO_WASM_RUNTIMES)
-   set(WRAP_MAIN "-Wl,-wrap=main")
+   target_link_options(eosio_testlib INTERFACE "-Wl,-wrap=main")
```

Comment:
Not present in cmake 3.10.

More generally, it seems like a fair bit of complexity can be avoided on cmake 3.13+, which makes me wonder if it's best to wait until removal of ubuntu 18 support before these changes